x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select_option.tsx

```diff
@@ -5,56 +5,203 @@
+    </EuiPopover>
+  );
+};
+
+export const ModelSelectOption: React.FC<ModelSelectOptionProps> = ({
+  modelId,
+  title,
+  description,
+  license,
+  deploymentState,
+  deploymentStateReason,
+  modelDetailsPageUrl,
+  isPlaceholder,
+  checked,
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const onMenuButtonClick = () => setIsPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const { createModel, startModel } = useActions(ModelSelectLogic);
+  const { areActionButtonsDisabled } = useValues(ModelSelectLogic);
+
   return (
-    <EuiFlexGroup direction="column" gutterSize="xs">
-      <EuiFlexItem>
-        <EuiTitle size="xs">
-          <h4>{title ?? model.model_id}</h4>
-        </EuiTitle>
+    <EuiFlexGroup alignItems="center">
+      {/* Selection radio button */}
+      <EuiFlexItem grow={false} style={{ flexShrink: 0 }}>
+        <EuiRadio id={modelId} checked={checked === 'on'} onChange={() => null} />
       </EuiFlexItem>
-      <EuiFlexItem>
-        <EuiFlexGroup gutterSize="s" alignItems="center" justifyContent="flexEnd">
-          {title && (
-            <EuiFlexItem>
-              <EuiTextColor color="subdued">{model.model_id}</EuiTextColor>
-            </EuiFlexItem>
-          )}
-          <EuiFlexItem grow={false}>
-            <TrainedModelHealth
-              modelState={parseModelStateFromStats(model)}
-              modelStateReason={parseModelStateReasonFromStats(model)}
-            />
+      {/* Title, model ID, description, license */}
+      <EuiFlexItem style={{ overflow: 'hidden' }}>
+        <EuiFlexGroup direction="column" gutterSize="xs">
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <h4>{title}</h4>
+            </EuiTitle>
           </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiFlexGroup gutterSize="xs">
-              <EuiFlexItem>
-                <MLModelTypeBadge type={type} />
-              </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTextColor color="subdued">{modelId}</EuiTextColor>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiFlexGroup gutterSize="xs" alignItems="center">
+              {license && (
+                <EuiFlexItem grow={false}>
+                  <EuiBadge color="hollow">License: {license}</EuiBadge>
+                </EuiFlexItem>
+              )}
+              {description && (
+                <EuiFlexItem style={{ overflow: 'hidden' }}>
+                  <EuiText size="xs">
+                    <div className="eui-textTruncate" title={description}>
+                      {description}
+                    </div>
+                  </EuiText>
+                </EuiFlexItem>
+              )}
             </EuiFlexGroup>
           </EuiFlexItem>
         </EuiFlexGroup>
       </EuiFlexItem>
+      {/* Status indicator OR action button */}
```

Comment:
rather than a comment it would make more sense to move this to a component with a descriptive name.

But that maybe a personal preference thing, comments in JSX seem like a code smell to me, but I could be wrong there, @sphilipse ?